vitrina/datasets/views.py

```
@@ -1,15 +1,28 @@
+from django.core.exceptions import ObjectDoesNotExist
 from django.views.generic import ListView
 from django.views.generic.detail import DetailView
 
 from vitrina.datasets.models import Dataset
+from vitrina.orgs.models import Organization
 
 
 class DatasetListView(ListView):
     model = Dataset
-    queryset = Dataset.public.order_by('-published')
     template_name = 'vitrina/datasets/list.html'
     paginate_by = 20
 
+    def get_queryset(self):
+        datasets = Dataset.public.order_by('-published')
+        organization = None
+        if self.kwargs.get('slug'):
+            try:
+                organization = Organization.objects.get(slug=self.kwargs['slug'])
+            except ObjectDoesNotExist:
+                pass
+        if organization:
+            datasets = Dataset.public.filter(organization=organization)
```

Let's be consistent and check if this is organization datasets list by checking `request.resolver_match.url_name`, same way as it is done in templates. And if request URL is organization datasets, then we should throw 404 if organization not found.